core/pom.xml

```diff
@@ -446,11 +451,6 @@ THE SOFTWARE.
       <artifactId>spring-aop</artifactId>
       <version>${spring.version}</version>
     </dependency>
-    <dependency>
-      <groupId>xpp3</groupId>
```

Comment:
Are there any transient dependencies on this library in plugins?
There is a risk of breaking the binary compatibility of the core.

I also see there are usages in the XStream fork: https://github.com/search?q=org%3Ajenkinsci+Xpp3Driver&type=Code . Version is the same: https://github.com/jenkinsci/xstream/blob/d3a5bd43086303169f4737758bf89be2e5584a71/pom.xml#L716.

Due to the XStream dependency, probably it's safe to remove it from here

Reply:
we can do a `mvn dependency:list` before and after this PR to assess this @mikecirioli.